### DISCUSSION TITLE
Support Condition Criteria in Batch Check Request

### DISCUSSION THREAD

**Discussion by embesozzi** [openfga/community#388] [Ideas]
Currently, the [Batch Check call](https://openfga.dev/docs/getting-started/perform-check#03-calling-batch-check-api) does not support adding condition criteria, such as `allOf` (AND) or `anyOf` (OR), for authorization checks. 
So far, the workaround is to iterate through `response.results.correlation_id` and verify that the allowed attribute value is `true` for all.

It would be great to support this feature in the future as authorization policies become more complex.

**Comment by hello-caleb** [comment]
Hi @embesozzi! Thanks feature request and the explanation of your use case.  If you have any more detail to add regarding your use case, that's always appreciated, as it helps the team to prioritize work from the backlog. Keep the ideas coming!
